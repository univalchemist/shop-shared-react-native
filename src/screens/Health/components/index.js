/* istanbul ignore file */

export {
  CallToActionLink,
  CardContainer,
  CardLoader,
  ErrorCard,
  TipCard,
} from './widgets';

export { default as ProductRecommendation } from './ProductRecommendation';
export { default as LifestyleTipsModal } from './LifestyleTipsModal';
export { default as LifestyleResults } from './LifestyleResults';
export { default as HealthScoreHistory } from './HealthScoreHistory';
export { default as HealthScore } from './HealthScore';
export { default as HealthResultCard } from './HealthResultCard';
export { default as GeneralTips } from './GeneralTips';
export { default as FaceAgingResults } from './FaceAgingResults';
