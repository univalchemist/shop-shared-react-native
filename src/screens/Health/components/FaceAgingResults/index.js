import React, { useState } from 'react';
import { View, StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { Slider } from 'react-native-elements';
import {
  Box,
  Flex,
  Image,
  Loader,
  ScreenHeadingText,
  Text,
} from '@wrappers/components';
import theme from '@theme';
import PropTypes from 'prop-types';
import { useIntl } from '@wrappers/core/hooks';
import { ErrorCard } from '@screens/Health/components/widgets/ErrorCard.js';
import { CardContainer } from '@screens/Health/components/widgets/CardContainer';
import { categories, logAction } from '@store/analytics/trackingActions';

const styles = StyleSheet.create({
  container: { paddingHorizontal: 32, paddingBottom: 32 },
});

export const FaceAgingResults = ({ faceAging }) => {
  const intl = useIntl();
  const [selectedAgeIndex, setSelectedAgeIndex] = useState(0);
  return (
    <View style={styles.container}>
      {faceAging.isError ? (
        renderFaceAgingErrorPanel({ intl })
      ) : faceAging.faceAgingIsDone &&
        faceAging.expectedTotalResults === faceAging.currentTotalResults ? (
        renderFaceAgingResults(
          faceAging,
          selectedAgeIndex,
          setSelectedAgeIndex,
          intl,
        )
      ) : (
        <>
          <ScreenHeadingText mb={2}>
            {intl.formatMessage({ id: 'futureMeTitle' })}
          </ScreenHeadingText>
          <CardContainer
            theme={theme}
            justifyContent="center"
            color={theme.colors.white}
            flexGrow={1}
          >
            <Loader
              primary
              textAlign={'center'}
              loadingText={intl.formatMessage({
                id: 'health.healthScoreHistory.lifestyleImage.processing',
                defaultMessage:
                  'Analysing in progress. This may take a little while. Please do not navigate away.',
              })}
            />
          </CardContainer>
        </>
      )}
    </View>
  );
};

const renderFaceAgingResults = (
  faceAging,
  selectedAgeIndex,
  setSelectedAgeIndex,
  intl,
) => {
  let faceAges = [];
  if (faceAges.length === 0) {
    faceAges = Object.keys(faceAging.results).sort();
  }
  const selectedAge = faceAges[selectedAgeIndex];
  const selectedAgeForDisplay = selectedAge == '67' ? '65' : selectedAge;

  return faceAges.length > 0 ? (
    <>
      <ScreenHeadingText mb={2}>
        {intl.formatMessage({ id: 'futureMeTitle' })}
      </ScreenHeadingText>
      <Box>
        <Text
          accessible={true}
          accessibilityLabel={intl.formatMessage(
            {
              id: 'health.sectionTitle.faceAging',
            },
            { selectedAge: selectedAgeForDisplay },
          )}
          color={theme.colors.gray[0]}
          fontWeight={theme.fontWeights.bold}
          fontSize={16}
          lineHeight={22}
          letterSpacing={0.3}
          mt={8}
          mb={8}
        >
          {intl.formatMessage(
            {
              id: 'health.sectionTitle.faceAging',
            },
            { selectedAge: selectedAgeForDisplay },
          )}
        </Text>
        {renderFaceAgingImages(selectedAge, faceAging, intl)}
        <Box mt={16}>
          <Slider
            style={{ width: '100%', height: 40 }}
            step={1}
            minimumValue={0}
            maximumValue={faceAges.length - 1}
            thumbTintColor={theme.colors.black}
            minimumTrackTintColor={theme.colors.black}
            maximumTrackTintColor={theme.colors.gray[4]}
            onValueChange={faceAgeIndex => {
              setSelectedAgeIndex(faceAgeIndex);
            }}
            onSlidingComplete={() => {
              logAction({
                category: categories.LIFESTYLE_OVERVIEW,
                action: `Slider age ${selectedAgeForDisplay}`,
              });
            }}
          />
          <Flex flex={1} flexDirection="row" />
        </Box>
        <Text
          accessibilityLabel={intl.formatMessage({
            id: 'health.faceAgingDescription',
          })}
          accessible={true}
          fontSize={14}
          mt={8}
          mb={16}
        >
          {intl.formatMessage({ id: 'health.faceAgingDescription' })}
        </Text>
      </Box>
    </>
  ) : (
    <Box />
  );
};

const renderFaceAgingImages = (selectedAge, faceAging, intl) => {
  return (
    <>
      {Object.keys(faceAging.results).map((faceAge, index) => {
        return (
          <Flex
            key={index}
            flexDirection="row"
            style={faceAge === selectedAge ? {} : { display: 'none' }}
          >
            <Box flex={1} marginRight={8}>
              <Box>
                <Image
                  accessibilityLabel={intl.formatMessage({
                    id:
                      'health.accessibilityLabel.LivingHealthyLifestyleFaceAgingImage',
                  })}
                  accessible={true}
                  accessibilityRole={'image'}
                  height={210}
                  source={{ uri: faceAging.results[faceAge].healthy }}
                  borderRadius={5}
                />
                <Text
                  accessibilityLabel={intl.formatMessage({
                    id: 'health.LivingHealthyLifestyle',
                  })}
                  accessible={true}
                  fontSize={14}
                  paddingTop={8}
                >
                  {intl.formatMessage({ id: 'health.LivingHealthyLifestyle' })}
                </Text>
              </Box>
            </Box>
            <Box flex={1}>
              <Box>
                <Image
                  accessibilityLabel={intl.formatMessage({
                    id:
                      'health.accessibilityLabel.LivingCurrentLifestyleFaceAgingImage',
                  })}
                  accessible={true}
                  accessibilityRole={'image'}
                  height={210}
                  source={{ uri: faceAging.results[faceAge].unhealthy }}
                  borderRadius={5}
                />
                <Text
                  accessibilityLabel={intl.formatMessage({
                    id: 'health.LivingCurrentLifestyle',
                  })}
                  accessible={true}
                  fontSize={14}
                  paddingTop={8}
                >
                  {intl.formatMessage({ id: 'health.LivingCurrentLifestyle' })}
                </Text>
              </Box>
            </Box>
          </Flex>
        );
      })}
    </>
  );
};

const renderFaceAgingErrorPanel = ({ intl }) => (
  <Box>
    <ScreenHeadingText mb={2}>
      {intl.formatMessage({ id: 'futureMeTitle' })}
    </ScreenHeadingText>
    <ErrorCard color={theme.colors.white} />
  </Box>
);

const mapStateToProps = ({ health: { faceAging } }) => ({
  faceAging,
});

const enhance = compose(connect(mapStateToProps));

export default enhance(FaceAgingResults);

FaceAgingResults.propTypes = {
  faceAging: PropTypes.shape({
    faceAgingIsDone: PropTypes.bool,
    expectedTotalResults: PropTypes.number,
    currentTotalResults: PropTypes.number,
    results: PropTypes.shape({}),
  }),
};
