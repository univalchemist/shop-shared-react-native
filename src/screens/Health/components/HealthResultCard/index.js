import React from 'react';
import PropTypes from 'prop-types';
import {
  healthIndicatorGreen,
  healthIndicatorPurple,
  healthIndicatorOrange,
  healthIndicatorRed,
  alcoholHighRisk,
  alcoholLowRisk,
  exerciseHighRisk,
  exerciseLowRisk,
  smokingLowRisk,
  smokingHighRisk,
  sleepLowRisk,
  sleepHighRisk,
  nutritionLowRisk,
  nutritionHighRisk,
  nutritionModerateRisk,
  mentalHealthLowRisk,
  mentalHealthHighRisk,
  mentalHealthModerateRisk,
} from '@images';
import { StyleSheet, View } from 'react-native';
import { Card } from 'react-native-elements';
import { Text, Image } from '@wrappers/components';
import theme from '@theme';
import { FormattedMessage } from 'react-intl';

const styles = StyleSheet.create({
  cardStyle: {
    width: 200,
    display: 'flex',
    borderRadius: 4,
    margin: 5,
    flex: 1,
    borderColor: 'white',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.1,
    shadowRadius: 3.84,

    elevation: 5,
  },
  indicatorStyles: {
    justifyContent: 'center',
    display: 'flex',
    flexDirection: 'row',
    paddingTop: 8,
    alignContent: 'space-between',
  },
  labelStyle: {
    fontWeight: '600',
    paddingLeft: 7,
    letterSpacing: 1.6,
    fontSize: theme.fontSizes[0],
    lineHeight: 16,
  },
  healthStatusStyles: {
    paddingTop: 19,
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
  },
  healthContextualDataStyles: {
    paddingBottom: 12,
  },
  scoreStyles: {
    fontSize: theme.fontSizes[5],
    fontWeight: 'bold',
    lineHeight: theme.lineHeights[5],
    letterSpacing: -1.5,
    height: 40,
  },
  healthImageStyle: {
    height: 40,
  },
  contextualDataStyles: {
    fontSize: theme.fontSizes[1],
    color: theme.colors.gray[1],
  },
  classStyles: {
    fontSize: theme.fontSizes[1],
    fontWeight: 'bold',
    lineHeight: theme.lineHeights[4],
    letterSpacing: 0.05,
    textAlign: 'center',
    minHeight: 48,
  },
});

const healthImages = {
  smokingLowRisk: smokingLowRisk,
  smokingHighRisk: smokingHighRisk,
  alcoholLowRisk: alcoholLowRisk,
  alcoholHighRisk: alcoholHighRisk,
  exerciseLowRisk: exerciseLowRisk,
  exerciseHighRisk: exerciseHighRisk,
  sleepHighRisk: sleepHighRisk,
  sleepLowRisk: sleepLowRisk,
  nutritionHighRisk: nutritionHighRisk,
  nutritionLowRisk: nutritionLowRisk,
  nutritionModerateRisk: nutritionModerateRisk,
  mentalHealthHighRisk: mentalHealthHighRisk,
  mentalHealthLowRisk: mentalHealthLowRisk,
  mentalHealthModerateRisk: mentalHealthModerateRisk,
};

const healthIndicatorColor = {
  Healthy: healthIndicatorGreen,
  Overweight: healthIndicatorPurple,
  Obese: healthIndicatorOrange,
  ExtremelyObese: healthIndicatorRed,
  LowRisk: healthIndicatorGreen,
  HighRisk: healthIndicatorOrange,
  VeryHighRisk: healthIndicatorRed,
  smokingLowRisk: healthIndicatorGreen,
  smokingHighRisk: healthIndicatorRed,
  alcoholLowRisk: healthIndicatorGreen,
  alcoholHighRisk: healthIndicatorRed,
  exerciseLowRisk: healthIndicatorGreen,
  exerciseHighRisk: healthIndicatorRed,
  sleepLowRisk: healthIndicatorGreen,
  sleepHighRisk: healthIndicatorRed,
  nutritionHighRisk: healthIndicatorRed,
  nutritionLowRisk: healthIndicatorGreen,
  nutritionModerateRisk: healthIndicatorOrange,
  mentalHealthHighRisk: healthIndicatorRed,
  mentalHealthLowRisk: healthIndicatorGreen,
  mentalHealthModerateRisk: healthIndicatorOrange,
};

const HealthResultCard = ({
  healthLabel,
  healthScore,
  healthClass,
  healthContextualData,
}) => (
  <Card containerStyle={styles.cardStyle}>
    <View style={styles.indicatorStyles}>
      <Image
        source={healthIndicatorColor[healthClass]}
        resizeMode="contain"
        width={18}
        height={18}
      />
      <Text style={styles.labelStyle}>{healthLabel}</Text>
    </View>
    <View style={styles.healthStatusStyles}>
      {healthScore ? (
        <Text style={styles.scoreStyles}>{healthScore}</Text>
      ) : (
        <Image
          style={styles.healthImageStyle}
          source={healthImages[healthClass]}
          resizeMode="contain"
          height={18}
        />
      )}
      <Text style={styles.classStyles}>
        <FormattedMessage id={`health.class.${healthClass}`} />
      </Text>
    </View>
    <View style={styles.healthContextualDataStyles}>
      <Text style={styles.contextualDataStyles}>{healthContextualData}</Text>
    </View>
  </Card>
);

HealthResultCard.propTypes = {
  healthLabel: PropTypes.string.isRequired,
  healthScore: PropTypes.number,
  healthClass: PropTypes.string.isRequired,
  healthContextualData: PropTypes.string.isRequired,
};

export default HealthResultCard;
