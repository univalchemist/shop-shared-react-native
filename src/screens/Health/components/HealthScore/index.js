import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import HealthScoreError from './HealthScoreError';
import HealthScoreDisplay, { HealthScoreLoading } from './HealthScoreDisplay';
import { PromiseStatus } from '@middlewares';
import { Card } from '@wrappers/components';
import { getUserHealthScore } from '@store/health/actions';

export const HealthScore = ({ healthScore = {}, getUserHealthScore }) => {
  useEffect(() => {
    const fetchUserHealthScore = async () => {
      await getUserHealthScore();
    };
    fetchUserHealthScore();
  }, [getUserHealthScore]);

  const { score, status } = healthScore;
  const isValidScore = !isNaN(parseInt(score, 10));
  const showHealthScore = () => {
    if (status === PromiseStatus.START) {
      return <HealthScoreLoading />;
    }
    if (status === PromiseStatus.ERROR || !isValidScore) {
      return <HealthScoreError />;
    }
    if (status === PromiseStatus.SUCCESS) {
      return <HealthScoreDisplay score={score} />;
    }
    return null;
  };

  return (
    <Card bg="white" flexGrow={1} px={4} pt={4} pb={24}>
      {showHealthScore()}
    </Card>
  );
};

const mapStateToProps = ({ health: { healthScore } }) => ({
  healthScore,
});

export default connect(mapStateToProps, { getUserHealthScore })(HealthScore);
