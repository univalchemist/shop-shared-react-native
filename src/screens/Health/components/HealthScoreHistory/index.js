import React, { useEffect } from 'react';
import HealthScoreHistoryGraph, {
  HealthScoreHistoryLoading,
} from './HealthScoreHistoryGraph';
import { connect } from 'react-redux';
import { PromiseStatus } from '@middlewares';
import { getUserHealthScoreHistory } from '@store/health/actions';
import HealthScoreHistoryError from './HealthScoreHistoryError';
import { Card } from '@wrappers/components';

export const HealthScoreHistory = ({
  healthScoreHistory = {},
  getUserHealthScoreHistory,
}) => {
  useEffect(() => {
    const fetchUserHealthScoreHistory = async () => {
      await getUserHealthScoreHistory();
    };
    fetchUserHealthScoreHistory();
  }, [getUserHealthScoreHistory]);

  const { data, status } = healthScoreHistory;
  const showHealthScoreHistory = () => {
    if (status === PromiseStatus.START) {
      return <HealthScoreHistoryLoading />;
    }
    if (status === PromiseStatus.ERROR) {
      return <HealthScoreHistoryError />;
    }
    if (status === PromiseStatus.SUCCESS) {
      return <HealthScoreHistoryGraph data={data} />;
    }
    return null;
  };

  return (
    <Card bg="white" flexGrow={1} justifyContent="center" px={3} pt={3} pb={3}>
      {showHealthScoreHistory()}
    </Card>
  );
};

const mapStateToProps = ({ health: { healthScoreHistory } }) => ({
  healthScoreHistory,
});

export default connect(mapStateToProps, { getUserHealthScoreHistory })(
  HealthScoreHistory,
);
