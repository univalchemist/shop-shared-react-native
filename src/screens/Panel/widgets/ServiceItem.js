import React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity } from 'react-native';
import { Box, Text } from '@wrappers/components';
import { Image } from '@wrappers/components';
import { useIntl, useTheme } from '@wrappers/core/hooks';
import styled from 'styled-components/native';

const StyledImage = styled(Image)`
  width: 100%;
  height: 100%;
`;

const ServiceItem = ({ image, titleId, descId, routeName, navigation }) => {
  const intl = useIntl();
  const theme = useTheme();
  const handlePress = () => {
    routeName && navigation.navigate(routeName);
  };

  return (
    <TouchableOpacity onPress={handlePress}>
      <Box
        flexDirection="row"
        justifyContent="space-between"
        height={134}
        backgroundColor={theme.colors.white}
        mb={16}
      >
        <Box flex={0.3}>
          <StyledImage source={image} />
        </Box>
        <Box flex={0.7} py={12} px={12}>
          <Text color={theme.colors.black}>
            {intl.formatMessage({
              id: titleId,
            })}
          </Text>
          <Text fontSize={14} color={theme.colors.gray[8]}>
            {intl.formatMessage({
              id: descId,
            })}
          </Text>
        </Box>
      </Box>
    </TouchableOpacity>
  );
};

ServiceItem.propTypes = {
  image: PropTypes.any,
  titleId: PropTypes.string,
  descId: PropTypes.string,
  routeName: PropTypes.string,
  navigation: PropTypes.object,
};

export default ServiceItem;
