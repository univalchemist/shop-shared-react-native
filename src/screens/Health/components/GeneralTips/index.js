import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { GeneralTipsCarousel } from './GeneralTipsCarousel';
import { GeneralTipsLoader } from './GeneralTipsLoader';
import { PromiseStatus } from '@middlewares';
import { Dimensions } from 'react-native';
import { useTheme } from '@wrappers/core/hooks';
import { ErrorCard } from '@screens/Health/components/widgets/ErrorCard';
import { Box, Text } from '@wrappers/components';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { FormattedMessage } from 'react-intl';
import { fetchLifestyleTips } from '@store/health/actions';

export const GeneralTips = ({ lifestyleTips, fetchLifestyleTips, locale }) => {
  const theme = useTheme();
  const horizontalMarginBetweenCards = theme.space[3];
  const viewportWidth = Dimensions.get('window').width;
  const viewportPadding = theme.space[4];
  const cardWidth = viewportWidth - viewportPadding * 2;
  const cardWidthWithSpacing = cardWidth + horizontalMarginBetweenCards;
  const cardHeight = 252.5;
  const viewportPaddingInsideCard = viewportPadding + 24;
  const loaderWidthInsideCard = viewportWidth - viewportPaddingInsideCard * 2;

  useEffect(() => {
    fetchLifestyleTips();
  }, [locale]);

  const renderGeneralTips = () => {
    switch (lifestyleTips.status) {
      case PromiseStatus.SUCCESS: {
        return lifestyleTips.tips.general ? (
          <GeneralTipsCarousel
            data={lifestyleTips.tips.general}
            cardWidthWithSpacing={cardWidthWithSpacing}
            cardHeight={cardHeight}
            horizontalMarginBetweenCards={horizontalMarginBetweenCards}
            cardWidth={cardWidth}
            viewportWidth={viewportWidth}
          />
        ) : (
          <Box
            width={cardWidthWithSpacing}
            height={cardHeight}
            px={horizontalMarginBetweenCards / 3}
            marginTop={16}
            alignSelf="center"
          >
            <ErrorCard color={'#FFF'} />
          </Box>
        );
      }
      case PromiseStatus.ERROR: {
        return (
          <Box
            width={cardWidthWithSpacing}
            height={cardHeight}
            px={horizontalMarginBetweenCards / 3}
            marginTop={16}
            alignSelf="center"
          >
            <ErrorCard color={'#FFF'} />
          </Box>
        );
      }
      default: {
        return (
          <GeneralTipsLoader
            cardWidthWithSpacing={cardWidthWithSpacing}
            cardHeight={cardHeight}
            horizontalMarginBetweenCards={horizontalMarginBetweenCards}
            cardWidth={cardWidth}
            viewportWidth={viewportWidth}
            loaderWidthInsideCard={loaderWidthInsideCard}
          />
        );
      }
    }
  };

  return (
    <>
      <Box>
        <Text
          color="gray.0"
          fontWeight={theme.fontWeights.bold}
          fontSize={16}
          lineHeight={22}
          letterSpacing={0.3}
          ml={32}
          mt={40}
        >
          <FormattedMessage id="health.sectionTitle.generalTips" />
        </Text>
      </Box>
      {renderGeneralTips()}
    </>
  );
};

const mapStateToProps = ({ health: { lifestyleTips }, intl: { locale } }) => ({
  lifestyleTips,
  locale,
});
const mapDispatchToProps = { fetchLifestyleTips };
const enhance = compose(connect(mapStateToProps, mapDispatchToProps));

GeneralTips.propTypes = {
  lifestyleTips: PropTypes.shape({
    tips: PropTypes.shape({ general: PropTypes.array }),
    status: PropTypes.string,
  }),
};

export default enhance(GeneralTips);
