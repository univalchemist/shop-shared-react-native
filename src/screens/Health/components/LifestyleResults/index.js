import { LifestyleResultsCarousel } from './LifestyleResultsCarousel';
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { Dimensions } from 'react-native';
import { Box, Text } from '@wrappers/components';
import { useIntl, useTheme } from '@wrappers/core/hooks';
import { LifestyleResultsLoader } from './LifestyleResultsLoader';
import { PromiseStatus } from '@middlewares';
import { getLifestyleResults } from './util';
import { injectIntl, FormattedMessage } from 'react-intl';
import { ErrorCard } from '@screens/Health/components/widgets/ErrorCard';

export const LifestyleResults = ({
  lifestyleResults,
  fetchUserLifestyleResultsStatus,
  navigation,
}) => {
  const theme = useTheme();
  const intl = useIntl();
  const horizontalMarginBetweenCards = theme.space[3];
  const viewportWidth = Dimensions.get('window').width;
  const viewportPadding = theme.space[4];
  const cardWidth = viewportWidth - viewportPadding * 2;
  const cardWidthWithSpacing = cardWidth + horizontalMarginBetweenCards;
  const cardHeight = 252.5;

  const renderLifestyleResults = () => {
    switch (fetchUserLifestyleResultsStatus) {
      case PromiseStatus.SUCCESS: {
        const lifestyleResultsArray = getLifestyleResults(
          lifestyleResults,
          theme,
          intl,
        );

        return (
          <LifestyleResultsCarousel
            data={lifestyleResultsArray}
            cardWidthWithSpacing={cardWidthWithSpacing}
            cardHeight={cardHeight}
            horizontalMarginBetweenCards={horizontalMarginBetweenCards}
            cardWidth={cardWidth}
            viewportWidth={viewportWidth}
            navigation={navigation}
          />
        );
      }
      case PromiseStatus.ERROR: {
        return (
          <Box
            width={cardWidthWithSpacing}
            height={cardHeight}
            px={horizontalMarginBetweenCards / 3}
            marginTop={16}
            alignSelf="center"
          >
            <ErrorCard color={theme.colors.error[3]} />
          </Box>
        );
      }
      default: {
        return (
          <LifestyleResultsLoader
            cardWidthWithSpacing={cardWidthWithSpacing}
            cardHeight={cardHeight}
            horizontalMarginBetweenCards={horizontalMarginBetweenCards}
          />
        );
      }
    }
  };

  return (
    <>
      <Box>
        <Text
          accessible={true}
          accessibilityLabel={intl.formatMessage({
            id: 'health.sectionTitle.lifestyleResults',
          })}
          color={theme.colors.gray[0]}
          fontWeight={theme.fontWeights.bold}
          fontSize={16}
          lineHeight={22}
          letterSpacing={0.3}
          ml={32}
          mt={40}
        >
          <FormattedMessage id="health.sectionTitle.lifestyleResults" />
        </Text>
      </Box>
      {renderLifestyleResults()}
    </>
  );
};

const mapStateToProps = ({
  health: { lifestyleResults, fetchUserLifestyleResultsStatus },
}) => ({
  lifestyleResults,
  fetchUserLifestyleResultsStatus,
});

const enhance = compose(connect(mapStateToProps, {}), injectIntl);

LifestyleResults.propTypes = {
  fetchUserLifestyleResultsStatus: PropTypes.string,
  lifestyleResults: PropTypes.shape({
    bmiScore: PropTypes.number,
    bmiClass: PropTypes.string,
  }),
};

export default enhance(LifestyleResults);
