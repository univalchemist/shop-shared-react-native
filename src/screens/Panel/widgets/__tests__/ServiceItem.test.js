import React from 'react';
import { TouchableOpacity } from 'react-native';
import { fireEvent } from 'react-native-testing-library';
import ServiceItem from '../ServiceItem';
import { renderForTest } from '@testUtils';
import { Image, Text } from '@wrappers/components';
import { axaPanelClinicImg } from '@images/doctor_landing';

describe('ServiceItem', () => {
  it('should render correctly', () => {
    const props = {
      image: axaPanelClinicImg,
      titleId: 'doctorLanding.service.panelClinic.title',
      descId: 'doctorLanding.service.panelClinic.desc',
      routeName: 'PANEL_SEARCH',
    };

    const serviceItem = renderForTest(<ServiceItem {...props} />);

    const image = serviceItem.getByType(Image);
    const texts = serviceItem.getAllByType(Text);

    expect(image).toBeDefined();
    expect(texts.length).toEqual(2);
  });

  it('should navigate to routeName when routeName is exist', () => {
    const props = {
      image: axaPanelClinicImg,
      titleId: 'doctorLanding.service.panelClinic.title',
      descId: 'doctorLanding.service.panelClinic.desc',
      routeName: 'PANEL_SEARCH',
    };
    const navigation = {
      navigate: jest.fn(),
    };

    const serviceItem = renderForTest(
      <ServiceItem {...props} navigation={navigation} />,
    );
    const clikableItem = serviceItem.getByType(TouchableOpacity);

    fireEvent(clikableItem, 'press');

    expect(navigation.navigate).toHaveBeenCalledWith(props.routeName);
  });

  it('should not navigate to routeName when routeName is not exist', () => {
    const props = {
      image: axaPanelClinicImg,
      titleId: 'doctorLanding.service.panelClinic.title',
      descId: 'doctorLanding.service.panelClinic.desc',
    };
    const navigation = {
      navigate: jest.fn(),
    };

    const serviceItem = renderForTest(
      <ServiceItem {...props} navigation={navigation} />,
    );

    const clikableItem = serviceItem.getByType(TouchableOpacity);

    fireEvent(clikableItem, 'press');

    expect(serviceItem).toMatchSnapshot();
  });
});
