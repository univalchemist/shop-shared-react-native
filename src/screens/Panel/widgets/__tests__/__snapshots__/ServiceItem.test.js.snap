// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceItem should not navigate to routeName when routeName is not exist 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          backgroundColor="#ffffff"
                          flexDirection="row"
                          height={134}
                          justifyContent="space-between"
                          mb={16}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#ffffff",
                                "flexDirection": "row",
                                "height": 134,
                                "justifyContent": "space-between",
                                "marginBottom": 16,
                              },
                            ]
                          }
                        >
                          <View
                            flex={0.3}
                            style={
                              Array [
                                Object {
                                  "flexBasis": 0,
                                  "flexGrow": 0.3,
                                  "flexShrink": 1,
                                },
                              ]
                            }
                          >
                            <Image
                              source={
                                Object {
                                  "testUri": "../../../src/images/doctor_landing/axa_panel_clinic.png",
                                }
                              }
                              style={
                                Array [
                                  Object {},
                                  Object {
                                    "height": "100%",
                                    "width": "100%",
                                  },
                                ]
                              }
                            />
                          </View>
                          <View
                            flex={0.7}
                            px={12}
                            py={12}
                            style={
                              Array [
                                Object {
                                  "flexBasis": 0,
                                  "flexGrow": 0.7,
                                  "flexShrink": 1,
                                  "paddingBottom": 12,
                                  "paddingLeft": 12,
                                  "paddingRight": 12,
                                  "paddingTop": 12,
                                },
                              ]
                            }
                          >
                            <Text
                              color="#000000"
                              fontFamily="default"
                              fontSize={16}
                              lineHeight={22}
                              style={
                                Array [
                                  Object {
                                    "color": "#000000",
                                    "fontFamily": "Arial",
                                    "fontSize": 16,
                                    "lineHeight": 22,
                                  },
                                ]
                              }
                            >
                              AXA Panel Clinics
                            </Text>
                            <Text
                              color="#666666"
                              fontFamily="default"
                              fontSize={14}
                              lineHeight={22}
                              style={
                                Array [
                                  Object {
                                    "color": "#666666",
                                    "fontFamily": "Arial",
                                    "fontSize": 14,
                                    "lineHeight": 22,
                                  },
                                ]
                              }
                            >
                              Find panel clinics and easily book an appointment with them.
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
